include/glow/Optimizer/GraphOptimizer/Pipeline/Pipeline.h

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_GRAPHOPTIMIZER_GRAPHOPTIMIZER_PIPELINES_PIPELINES_H
+#define GLOW_OPTIMIZER_GRAPHOPTIMIZER_GRAPHOPTIMIZER_PIPELINES_PIPELINES_H
+
+#include "glow/Optimizer/GraphOptimizer/CompilationContext.h"
+#include "glow/Support/Support.h"
+
+#include <bitset>
+
+namespace glow {
+
+/// Define an enum to identify all FunctionPasses, used to declare inside a
+/// FunctionPassConfig for a Pipeline.
+enum class FunctionPassID {
+#define FUN_PASS(PASS_NAME) PASS_NAME,
+#include "glow/Optimizer/GraphOptimizer/FunctionPasses.def"
+};
+
+/// Specifies convergence mode for a FunctionPass.
+enum class ConvergenceMode {
+  OnePass,         // Run a single pass over the Function.
```

Comment:
Very minor nit: We usually put comments before declarations, i.e. on the previous line.